Client/Frontend/Browser/LinkPreviewViewController.swift

```diff
@@ -9,33 +9,49 @@ import Shared
 
 class LinkPreviewViewController: UIViewController {
 
-  let url: URL
+  private let url: URL
+  private weak var parentTab: Tab?
+  private var currentTab: Tab?
+  private weak var browserController: BrowserViewController?
 
-  init(url: URL) {
+  init(url: URL, for tab: Tab, browserController: BrowserViewController) {
```

Comment:
Let's fix the tight coupling in a rewrite later this year, ticket for it here https://github.com/brave/brave-ios/issues/5934